### PR TITLE
Longest Task First with task split up logic and UI updates

### PR DESCRIPTION
**Pull request by hollyyuqizheng**
Main changes from the current longest task first scheduler:
- Added a "isCompletelyScheduled" tag to ScheduledTask class. This tag indicates whether this task is completely or only partially scheduled
- Modified LongestTaskScheduler class to set this completeness tag for each task
- Modified tests for longest and shortest task scheduler to reflect this new tag
- Modified request.js to display a warning on the result scheduled task card on the UI if the task is not completely scheduled 

Screenshot 2020-07-29 at 10 53 52 AM

### CODE REVIEW

**Review by talvarez0507 — APPROVED** [review]

**Review by raulcruise — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/data/ScheduledTask.java [hunk 1]
```diff
@@ -37,4 +39,8 @@ public boolean equals(Object other) {
   public static boolean equals(ScheduledTask a, ScheduledTask b) {
     return a.task.equals(b.task) && a.startTime.equals(b.startTime);
   }
+
+  public void setCompleteness(boolean scheduleStatus) {
```
Comment (jamesnam-google) [inline portfolio/src/main/java/com/google/sps/data/ScheduledTask.java]: Consider renaming scheduledStatus to isCompletelyScheduled.

Then in the method content you can do,

this.isCompl... = isCompl...

File: portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java [hunk 2]
```diff
@@ -159,7 +162,10 @@ private TimeRange scheduleTaskSegment(
       TimeRangeGroup availableTimesGroup) {
     Duration taskDuration = task.getDuration();
 
-    ScheduledTask scheduledTask = new ScheduledTask(task, scheduledTime);
+    // Assumes this task can be scheduled for now.
+    // If not, the true tag will be overwritten at the end of the scheduleOneTask method.
+    ScheduledTask scheduledTask =
+        new ScheduledTask(task, scheduledTime, /* isCompletelyScheduled= */ true);
```
Comment (jamesnam-google) [inline portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java]: Consider passing Optional.empty() for completely scheduled param. That way, we can avoid accidentally keeping this placeholder param.

**Review by jamesnam-google — APPROVED** [review]
Approving the change with comments

**Review by rgcase — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/data/ScheduledTask.java (hunk 1 above)
Comment (hollyyuqizheng) [reply portfolio/src/main/java/com/google/sps/data/ScheduledTask.java]: Updated.

File: portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java (hunk 2 above)
Comment (hollyyuqizheng) [reply portfolio/src/main/java/com/google/sps/data/LongestTaskFirstScheduler.java]: Updated.

**Pull request merged**